src/sql/workbench/contrib/resourceViewer/browser/resourceViewerEditor.ts

```diff
@@ -0,0 +1,176 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Taskbar } from 'sql/base/browser/ui/taskbar/taskbar';
+import { IDisposable } from 'vs/base/common/lifecycle';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+import { IContextKeyService, IContextKey, RawContextKey } from 'vs/platform/contextkey/common/contextkey';
+import { CancellationToken } from 'vs/base/common/cancellation';
+import { IStorageService } from 'vs/platform/storage/common/storage';
+import * as DOM from 'vs/base/browser/dom';
+import { BaseEditor } from 'vs/workbench/browser/parts/editor/baseEditor';
+import { EditorOptions } from 'vs/workbench/common/editor';
+import { ITelemetryService } from 'vs/platform/telemetry/common/telemetry';
+import { IWorkbenchThemeService } from 'vs/workbench/services/themes/common/workbenchThemeService';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { IResourceViewerStateChangedEvent } from 'sql/workbench/common/editor/resourceViewer/resourceViewerState';
+import { ResourceViewerInput } from 'sql/workbench/browser/editor/resourceViewer/resourceViewerInput';
+import { ResourceViewerTable } from 'sql/workbench/contrib/resourceViewer/browser/resourceViewerTable';
+
+export const CONTEXT_RESOURCE_VIEWER_EDITOR = new RawContextKey<boolean>('inResourceViewerEditor', false);
+
+export interface ResourceViewerTableViewState {
+	scrollTop: number;
+	scrollLeft: number;
+}
+
+export class ResourceViewerEditor extends BaseEditor {
+	public static readonly ID: string = 'workbench.editor.resource-viewer';
+
+	private _container!: HTMLElement;
+	private _actionBar!: Taskbar;
+	private _resourceViewerTable!: ResourceViewerTable;
+	private _stateListener: IDisposable | undefined;
+	private _columnChangeListener: IDisposable | undefined;
+	private _rowCountChangeListener: IDisposable | undefined;
+	private _filterStateChangeListener: IDisposable | undefined;
+
+	private _resourceViewerEditorContextKey: IContextKey<boolean>;
+
+	private _savedTableViewStates = new Map<ResourceViewerInput, ResourceViewerTableViewState>();
+
+	constructor(
+		@ITelemetryService telemetryService: ITelemetryService,
+		@IWorkbenchThemeService themeService: IWorkbenchThemeService,
+		@IInstantiationService private _instantiationService: IInstantiationService,
+		@IContextKeyService private _contextKeyService: IContextKeyService,
+		@IEditorService editorService: IEditorService,
+		@IStorageService storageService: IStorageService
+	) {
+		super(ResourceViewerEditor.ID, telemetryService, themeService, storageService);
+		this._resourceViewerEditorContextKey = CONTEXT_RESOURCE_VIEWER_EDITOR.bindTo(this._contextKeyService);
+
+		if (editorService) {
```

Comment:
You also shouldn't need this. This is only in other editors as a hack.

Reply:
Mind explaining this a bit - what's the hack needed for in their case?